syft/pkg/cataloger/cpe.go

```diff
@@ -82,37 +83,115 @@ func candidateVendors(p pkg.Package) []string {
 		vendors = append(vendors, fmt.Sprintf("python-%s", p.Name))
 	case pkg.Java:
 		if p.MetadataType == pkg.JavaMetadataType {
-			if metadata, ok := p.Metadata.(pkg.JavaMetadata); ok && metadata.PomProperties != nil {
-				// derive the vendor from the groupID (e.g. org.sonatype.nexus --> sonatype)
-				if strings.HasPrefix(metadata.PomProperties.GroupID, "org.") || strings.HasPrefix(metadata.PomProperties.GroupID, "com.") {
-					fields := strings.Split(metadata.PomProperties.GroupID, ".")
-					if len(fields) >= 3 {
-						vendors = append(vendors, fields[1])
-					}
-				}
-			}
+			vendors = append(vendors, candidateVendorsForJava(p)...)
 		}
 	}
 	return vendors
 }
 
 func candidateProducts(p pkg.Package) []string {
-	var products = []string{p.Name}
-	switch p.Language {
-	case pkg.Java:
-		if p.MetadataType == pkg.JavaMetadataType {
-			if metadata, ok := p.Metadata.(pkg.JavaMetadata); ok && metadata.PomProperties != nil {
-				// derive the product from the groupID (e.g. org.sonatype.nexus --> nexus)
-				if strings.HasPrefix(metadata.PomProperties.GroupID, "org.") || strings.HasPrefix(metadata.PomProperties.GroupID, "com.") {
-					fields := strings.Split(metadata.PomProperties.GroupID, ".")
-					if len(fields) >= 3 {
-						products = append(products, fields[2])
-					}
-				}
-			}
+	products := []string{p.Name}
+	if p.Language == pkg.Java {
+		products = append(products, candidateProductsForJava(p)...)
+	}
+	return products
+}
+
+func candidateProductsForJava(p pkg.Package) []string {
+	var products []string
+	if p.MetadataType == pkg.JavaMetadataType {
+		if product := productFromPomPropertiesGroupID(p); product != "" {
+			products = append(products, product)
 		}
-	default:
-		return products
 	}
 	return products
 }
+
+func candidateVendorsForJava(p pkg.Package) []string {
+	var vendors []string
+	if p.MetadataType == pkg.JavaMetadataType {
+		if product := vendorFromPomPropertiesGroupID(p); product != "" {
```

Comment:
this is not product right? I think it is vendor? 
```suggestion
		if vendor := vendorFromPomPropertiesGroupID(p); vendor != "" {
```

Reply:
Yep, correct. I'm working on a revision to this branch that cleans some of this up and adds test, will push shortly